### PR TITLE
sys-libs/db: port to EAPI 7

### PR DESCRIPTION
**Pull request by hanetzer**
Continuation of https://github.com/gentoo/gentoo/pull/8314

Package-Manager: Portage-2.3.76, Repoman-2.3.17

### CODE REVIEW

**Comment by hanetzer** [comment]
Ah, didn't realize more recent versions had happened. Gimme a bit on that.

File: sys-libs/db/db-5.3.28-r4.ebuild [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+#PATCHNO=`echo ${PV}|sed -e "s,\(.*_p\)\([0-9]*\),\2,"`
+PATCHNO=${PV/*.*.*_p}
+if [[ ${PATCHNO} == "${PV}" ]] ; then
+	MY_PV=${PV}
+	MY_P=${P}
+	PATCHNO=0
+else
+	MY_PV=${PV/_p${PATCHNO}}
+	MY_P=${PN}-${MY_PV}
+fi
```
Comment (xdch47) [inline sys-libs/db/db-5.3.28-r4.ebuild]: why not using `ver_cut` ?

File: sys-libs/db/db-5.3.28-r4.ebuild [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+
+S="${WORKDIR}/${MY_P}"
+BUILD_DIR="${S}/build${MULTIBUILD_VARIANT}"
+DESCRIPTION="Oracle Berkeley DB"
+HOMEPAGE="http://www.oracle.com/technetwork/database/database-technologies/berkeleydb/overview/index.html"
+SRC_URI="http://download.oracle.com/berkeley-db/${MY_P}.tar.gz"
+for (( i=1 ; i<=${PATCHNO} ; i++ )) ; do
+	export SRC_URI="${SRC_URI} http://www.oracle.com/technology/products/berkeley-db/db/update/${MY_PV}/patch.${MY_PV}.${i}"
+done
```
Comment (xdch47) [inline sys-libs/db/db-5.3.28-r4.ebuild]: couldn'- the patches here directly been added to `PATCHES` array ?

Comment (hanetzer) [reply sys-libs/db/db-5.3.28-r4.ebuild]: Well, they do have to be here in order to download them. Could be directly added to the
patches array, but my goal is a minimal change to move to eapi7.

File: sys-libs/db/db-5.3.28-r4.ebuild (hunk 1 above)
Comment (hanetzer) [reply sys-libs/db/db-5.3.28-r4.ebuild]: To be frank I don't know quite enough of eapi7-ver/native eapi7 version functions
to change this, and I don't see much reason to. It works already, so why change
it? If there's something to be gained I'm down for that, though.

File: sys-libs/db/db-5.3.28-r4.ebuild (hunk 2 above)
Comment (xdch47) [reply sys-libs/db/db-5.3.28-r4.ebuild]: just I just had a look, as you mentioned in #8314 "patches seem to be formatted to be -p0"...
so patches array is not a good idea

File: sys-libs/db/db-5.3.28-r4.ebuild [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+	"${FILESDIR}"/${PN}-6.0.20-test-link.patch
+
+	# Needed when compiling with clang
+	"${FILESDIR}"/${PN}-5.1.29-rename-atomic-compare-exchange.patch
+)
+
+src_prepare() {
+	for (( i=1 ; i<=${PATCHNO} ; i++ ))
+	do
+		eapply "${DISTDIR}"/patch."${MY_PV}"."${i}"
```
Comment (xdch47) [inline sys-libs/db/db-5.3.28-r4.ebuild]: should that be eapply -p0 ?

File: sys-libs/db/db-5.3.28-r4.ebuild (hunk 1 above)
Comment (xdch47) [reply sys-libs/db/db-5.3.28-r4.ebuild]:
```
PATCHNO="$(ver_cut 5)"
MY_PV="$(ver_cut 1-3)"
MY_P=${PN}-${MY_PV}

S="${WORKDIR}/${MY_P}"
....

for (( i=1 ; i<=${PATCHNO:-0} ; i++ )) ; do
	SRC_URI+=" http://www.oracle.com/technology/products/berkeley-db/db/update/${MY_PV}/patch.${MY_PV}.${i}"
done

```

seems cleaner and easier for me - why porting to eapi7 without using its features ?

File: sys-libs/db/db-5.3.28-r4.ebuild (hunk 2 above)
Comment (xdch47) [inline sys-libs/db/db-5.3.28-r4.ebuild]: why `export` ? 
`SRC_URI+=" http://www.oracle/..." ` should be sufficient, or?

File: sys-libs/db/db-5.3.28-r4.ebuild (hunk 1 above)
Comment (xdch47) [reply sys-libs/db/db-5.3.28-r4.ebuild]: the patchname is also `quite generic`, maybe something like
` .../update/${MY_PV}/patch.${MY_PV}.${i} -> ${MY_P}_p${i}` would be more clear.

furthermore I think homepage und source uri could be changed to `https`

File: sys-libs/db/db-5.3.28-r4.ebuild (hunk 3 above)
Comment (hanetzer) [reply sys-libs/db/db-5.3.28-r4.ebuild]: Actually yeah, probably. None of the ebuild's I've touched have a _p element
so it didn't flag wrong.

File: sys-libs/db/db-5.3.28-r4.ebuild (hunk 2 above)
Comment (hanetzer) [reply sys-libs/db/db-5.3.28-r4.ebuild]: Dunno, original person did that, but yeah, that should in theory be enough.

File: sys-libs/db/db-5.3.28-r4.ebuild (hunk 1 above)
Comment (hanetzer) [reply sys-libs/db/db-5.3.28-r4.ebuild]: > ```
> PATCHNO="$(ver_cut 5)"
> MY_PV="$(ver_cut 1-3)"
> MY_P=${PN}-${MY_PV}
> 
> S="${WORKDIR}/${MY_P}"
> ....
> 
> for (( i=1 ; i<=${PATCHNO:-0} ; i++ )) ; do
> 	SRC_URI+=" https://www.oracle.com/technology/products/berkeley-db/db/update/${MY_PV}/patch.${MY_PV}.${i}"
> done
> ```
> 
> seems cleaner and easier for me - why porting to eapi7 without using its features ?

The major feature is the cross-friendlyness, but that does appear to be enough. I'll
refactor with your suggestions as soon as I'm at my workstation.

**Comment by ajakk** [comment]
Only masked versions are still EAPI<7. Is this still needed?

**Comment by arkamar** [comment]
Versions 5.3.28 and 6.0.35 has already EAPI 7 since commit 3fda85d3fef1 - `sys-libs/db: Another batch of EAPI-7 revbumps`. Versions 6.{1,2}.38 are masked, original ebuilds have changed, therefore it would require PR revision.

I would say this PR is obsolete.

**Pull request closed without merging**